mne/utils/numerics.py

```diff
@@ -703,7 +703,13 @@ def _sort_keys(x):
     return keys
 
 
-def object_diff(a, b, pre=''):
+def _array_equal_nan(a, b, ignore_nan):
+    a_idx = ~np.isnan(a) if ignore_nan else slice(None)
+    b_idx = ~np.isnan(b) if ignore_nan else slice(None)
+    return np.array_equal(a[a_idx], b[b_idx]) and np.array_equal(a_idx, b_idx)
```

Comment:
this does not pass the tests I added

Reply:
Thinking about our `object_diff` function I'm realizing that we want things like `[0, np.nan, np.inf]` to work, so actually I think we don't need the `ignore_nan` parameter (it can just always act like True).

> we can always compare a_idx and b_idx aswell

Yes, but we are now getting to the point where we are replicating `assert_array_equal`. It's simpler and safer (since `assert_array_equal` is very widely tested) for us just to use the try/except pattern directly.